piff/gp_interp_2pcf.py

```diff
@@ -0,0 +1,375 @@
+        """
+        X = np.array([self.getProperties(star) for star in stars])
+        y = np.array([star.fit.params for star in stars])
+        y_err = np.array([np.sqrt(np.diag(star.fit.params_cov)) for star in stars])
+        
+        self._X = X
+        self._y = y
+        self._y_err = np.sqrt(y_err**2 + self.white_noise**2)
+
+        if self.npca > 0:
+            from sklearn.decomposition import PCA
+            self._pca = PCA(n_components=self.npca, whiten=True)
+            self._pca.fit(y)
+            y = self._pca.transform(y)
+            if y_err is not None:
+                y_err = self._pca.transform(y_err)
+            self._y_pca, self._y_pca_err = y, y_err
+            self.nparams = self.npca
+
+        if self.normalize:
+            self._mean = np.mean(y,axis=0)
+        else:
+            self._mean = np.zeros(self.nparams)
+        self._init_theta = []
+        for i in range(self.nparams):
+            kernel = self.kernels[i]
+            self._init_theta.append(kernel.theta)
+            self.kernels[i] = self._fit(self.kernels[i],
+                                        X, y[:,i]-self._mean[i],
+                                        y_err=y_err[:,i], logger=logger)
+            if logger:
+                logger.info('param %d: %s',i,kernel.set_params())
+
+
+    def interpolate(self, star, logger=None):
+        """Perform the interpolation to find the interpolated parameter vector at some position.
+
+        :param star:        A Star instance to which one wants to interpolate
+        :param logger:      A logger object for logging debug info. [default: None]
+
+        :returns: a new Star instance with its StarFit member holding the interpolated parameters
+        """
+        # because of sklearn formatting, call interpolateList and take 0th entry
+        return self.interpolateList([star], logger=logger)[0]
+
+    def interpolateList(self, stars, logger=None):
+        """Perform the interpolation for a list of stars.
+
+        :param star_list:   A list of Star instances to which to interpolate.
+        :param logger:      A logger object for logging debug info. [default: None]
+
+        :returns: a list of new Star instances with interpolated parameters
+        """
+        Xstar = np.array([self.getProperties(star) for star in stars])
+        y = self._predict(Xstar)
+        fitted_stars = []
+        for y0, star in zip(y, stars):
+            if star.fit is None:
+                fit = StarFit(y)
+            else:
+                fit = star.fit.newParams(y0)
+            fitted_stars.append(Star(star.data, fit))
+        return fitted_stars
+
+    def _finish_write(self, fits, extname):
+        # Note, we're only storing the training data and hyperparameters here, which means the
+        # Cholesky decomposition will have to be re-computed when this object is read back from
+        # disk.
+
+        init_theta = np.array([self._init_theta[i] for i in range(self.nparams)])
```

Comment:
is there a reason for keeping the initial theta?

Reply:
It was to be consistent with the gp_interp.py, I have not a strong opinion on that. I guess like that you can write it explicitly in output file and keep a track of what was the input.